kotest-assertions/kotest-assertions-shared/src/jvmTest/kotlin/com/sksamuel/kotest/eq/IterableEqTest.kt

```diff
@@ -42,12 +63,126 @@ class IterableEqTest : FunSpec({
       }
    }
 
+   test("should not give error for kotlin ordered set comparison with list") {
+      val error = IterableEq.equals(setOf(1, 2, 3), listOf(1, 2, 3)).shouldBeNull()
```

Comment:
set and list containing equal elements should not be equal.

Reply:
I agree in principle, yes.  Full agreement.  At stake here is backward compatibility.  `setOf(something)` returns a `LinkedHashSet` which is an ordered set.  What is wrong and illegal to compare is, down to the metal, an ordered collection vs. a collection that has no ordering guarantee.  An ordered set like `LinkedHashSet` is a creature of convenience (sigh) that is ordered more often than not, and also a set (no duplicates).  The existing code does unordered comparison between `Set and Set`, and ordered comparison between `Set xor Set` so I punted.  Should we break backward compatibility?  That is going to create a very angry nest of wasps among practically minded users.  We can do either, there are good arguments for both.  The issue itself supports backward compatibility ...

Reply:
Backwards compatibility all the way.